### PR TITLE
Adds Typescript support, closes #6

### LINKED ISSUE #6: Add Typescript

### PR DESCRIPTION
**Pull request by mehlium**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
     "build": "rollup -c",
     "dev": "rollup -c -w",
     "start": "sirv public",
-    "push": "cp -r public/ build/ && kaios-scripts push"
+    "clean": "rm -r build/ public/build",
+    "push": "yarn clean && yarn build && cp -r public/ build/ && kaios-scripts push",
```
Comment (mehlium) [inline package.json:12]: Noticed we were not cleaning properly on push, should be fixed now.

File: .gitignore [hunk 2]
```diff
@@ -9,3 +9,4 @@ yarn-error.log
 /build/
 
 .DS_Store
+.vscode/
```
Comment (Citrullin) [inline .gitignore]: This shouldn't be in the repo. This is a user-specific configuration and should therefore be in the global gitignore.
https://davidwalsh.name/global-gitignore

Comment (Citrullin) [reply .gitignore]: Ohh, shit, there is already .idea, did I push that?

Comment (mehlium) [reply .gitignore]: No, I added .idea, vscode probably came from the tool. I'm cool with un-ignoring this!

**Pull request merged**